### PR TITLE
NUP-2310 Foundation for network visualization framework

### PR DESCRIPTION
**Pull request by oxtopus**
@scottpurdy @subutai This lays the foundation for visualizing networks and requires recent changes to nupic.bindings to allow for the introspection of network links (i.e. this will fail until the next release has completed).  With this in place, we can focus on iterating on the details of the visualization moving forward, but for now it has a few naive renderers.

### CODE REVIEW

**Comment by oxtopus** [comment]
@scottpurdy The doesn't block on user input, if that's what your suggesting.  Regardless, I'll move it into an examples dir.  Thanks for the suggestion.

**Comment by scottpurdy** [comment]
@oxtopus - yes, I did think that `viz.render` would pop up a window and wait for user input. What actually happens?

**Comment by oxtopus** [comment]
@scottpurdy I removed the test and added an example.  Please let me know if you have any additional comments to close out this PR, otherwise I will assume it's good to merge.

File: src/nupic/frameworks/viz/examples/visualize_network.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+  # Render w/ graphviz
+  viz.render(renderer=GraphVizRenderer)
+
+  # Render w/ networkx
+  viz.render(renderer=NetworkXRenderer)
+
+
+
+if __name__ == "__main__":
```
Comment (scottpurdy) [inline src/nupic/frameworks/viz/examples/visualize_network.py]: Alternatively, you could have the setup.py specify `main` in this file to be installed as an executable (like `nupic-network-visualization-example` or something). Up to you.

File: src/nupic/frameworks/viz/graphviz_renderer.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+
+
+class GraphVizRenderer(object):
+  """
+  Network visualization "renderer" implementation to render a network with
+  graphviz.
+  """
+  def render(self, G):
+    G = nx.nx_agraph.to_agraph(G)
```
Comment (scottpurdy) [inline src/nupic/frameworks/viz/graphviz_renderer.py]: `graph` or something would be more explanatory (here and elsewhere). And local variables should start with a lowercase letter

Comment (oxtopus) [inline src/nupic/frameworks/viz/graphviz_renderer.py]: Thanks.  All of the networkx documentation and related literature tend to use the `G` convention.  I will change to `graph` per your request.

**Pull request merged**